src/electron/routing_service.ts

```diff
@@ -213,7 +216,30 @@ export class RoutingDaemon {
     return this.disconnected;
   }
 
-  public set onNetworkChange(newListener: ((status: TunnelStatus) => void)|undefined) {
+  public set onNetworkChange(newListener: ((status: TunnelStatus) => void) | undefined) {
     this.networkChangeListener = newListener;
   }
+
+  private isDaemonInstalled(): Promise<boolean> {
+    return new Promise<boolean>(resolve => {
+      const connection: Socket = createConnection(SERVICE_NAME, () => connection.destroy());
+      connection.once("error", () => connection.destroy());
```

Comment:
Is close still called on error or will the Promise be pending forever?

Reply:
`destroy` will do the close thing.